include/zephyr/sensing/sensing_sensor.h

```diff
@@ -0,0 +1,594 @@
+ *
+ */
+typedef int (*sensing_sensor_read_sample_t)(
+		const struct device *dev,
+		void *buf, int size);
+
+/**
+ * @brief Sensor process data callback.
+ *
+ * Only virtual sensor need implement this callback.
+ * Virtual sensor can receive reporter's data and do fusion computing
+ * in this callback.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param reporter The reporter handle who delivered this sensor data
+ *
+ * @param buf The buffer stored the reporter's sensor data.
+ *
+ * @param size The size of the buffer in bytes.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+typedef int (*sensing_sensor_process_t)(
+		const struct device *dev,
+		int reporter,
+		void *buf, int size);
+
+/**
+ * @brief Trigger a sensor to do self calibration
+ *
+ * If not support self calibration, can not implement this callback.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+typedef int (*sensing_sensor_self_calibration_t)(
+		const struct device *dev);
+
+/**
+ * @brief Sensitivity arbitration callback.
+ *
+ * This callback API provides a chance for sensor to do customized arbitration on data change
+ * sensitivity.
+ * The sensor can check two sequential samples with client's sensitivity value (passed with
+ * parameters in this callback) and decide if can pass the sensor sample to its client.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param index The value fields index to be set, -1 for all fields (global).
+ *
+ * @param sensitivity The sensitivity value.
+ *
+ * @param last_sample_buf The buffer stored last sample data.
+ *
+ * @param last_sample_size The size of last sample's data buffer in bytes
+ *
+ * @param current_sample_buf The buffer stored current sample data.
+ *
+ * @param current_sample_size The size of current sample's data buffer in bytes
+ *
+ * @return 1 on test passed, 0 on test not passed, or negative error value on other errors.
+ *
+ */
+typedef int (*sensing_sensor_sensitivity_test_t)(
+		const struct device *dev,
+		int index, uint32_t sensitivity,
+		void *last_sample_buf, int last_sample_size,
+		void *current_sample_buf, int current_sample_size);
+
+/**
+ * @brief Set current report interval callback.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param value The value to be set.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+typedef int (*sensing_sensor_set_interval_t)(
+		const struct device *dev,
+		uint32_t value);
+
+/**
+ * @brief Get current report interval callback.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param value The data buffer to receive value.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+typedef int (*sensing_sensor_get_interval_t)(
+		const struct device *dev,
+		uint32_t *value);
+
+/**
+ * @brief Set data change sensitivity callback.
```

Comment:
Please check description for these functions, they seem outdated.

Reply:
removed callback